homeassistant/components/nuki/config_flow.py

```diff
@@ -110,7 +110,9 @@ async def async_step_reauth_confirm(self, user_input=None):
             errors["base"] = "unknown"
 
         if not errors:
-            existing_entry = await self.async_set_unique_id(info["ids"]["hardwareId"])
+            existing_entry = await self.async_set_unique_id(
+                hex(info["ids"]["hardwareId"]).split("x")[-1].upper()
```

Comment:
Maybe break out the bridge id parsing to a helper function that we can use in both places?